### PR TITLE
[AzureMonitorDistro] upgrade OpenTelemetry for hotfix

### PR DESCRIPTION
**Pull request by TimothyMothra**
I created this PR so our team can review these changes.

## Hotfix
I created a new `hotfix/azure.monitor.opentelemetry.aspnetcore_1.1.1`, forked from [tag/Azure.Monitor.OpenTelemetry.AspNetCore_1.1.0](https://github.com/Azure/azure-sdk-for-net/releases/tag/Azure.Monitor.OpenTelemetry.AspNetCore_1.1.0)
Our hotfix will release from this branch.
This PR is merging into this branch.

## Changes
This PR is upgrades the OpenTelemetry dependencies to 1.8.1.

### CODE REVIEW

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

**Review by jsquire — APPROVED** [review]
Central package versions look good.

**Review by vishweshbankwar — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Release History
 
+## 1.1.1 (Unreleased)
+
+* Update OpenTelemetry dependencies. This includes a hotfix for [GHSA-vh2m-22xx-q94f](https://github.com/open-telemetry/opentelemetry-dotnet/security/advisories/GHSA-vh2m-22xx-q94f).
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:5]: please include the change in the changelog itself esp. since it is a breaking change.

Comment (cijothomas) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:5]: also, mention how to get back the old behavior in the changelog itself.

**Comment by TimothyMothra** [comment]
Updated the changelog with the quote from the OTel changelog.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+  ([#43197](https://github.com/Azure/azure-sdk-for-net/pull/43197))
+  - OpenTelemetry 1.8.0
+  - OpenTelemetry.Extensions.Hosting 1.8.0
+  - OpenTelemetry.Instrumentation.AspNetCore 1.8.1
+  - OpenTelemetry.Instrumentation.Http 1.8.1
+  - This includes a hotfix for [GHSA-vh2m-22xx-q94f](https://github.com/open-telemetry/opentelemetry-dotnet/security/advisories/GHSA-vh2m-22xx-q94f) which redacts the query string values by default. 
+
+### Breaking Changes
+
+* OpenTelemetry.Instrumentation.AspNetCore and OpenTelemetry.Instrumentation.Http include a hotfix for [GHSA-vh2m-22xx-q94f](https://github.com/open-telemetry/opentelemetry-dotnet/security/advisories/GHSA-vh2m-22xx-q94f).
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:17]: nit: 
```suggestion
* `OpenTelemetry.Instrumentation.AspNetCore` and `OpenTelemetry.Instrumentation.Http` include a hotfix for [GHSA-vh2m-22xx-q94f](https://github.com/open-telemetry/opentelemetry-dotnet/security/advisories/GHSA-vh2m-22xx-q94f).
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+  By default any values detected in the query string component of requests are replaced with the text `Redacted` when building the `url.query` tag.
+  For example, `?key1=value1&key2=value2` becomes `?key1=Redacted&key2=Redacted`.
+  You can disable this redaction by setting the environment variable `OTEL_DOTNET_EXPERIMENTAL_ASPNETCORE_DISABLE_URL_QUERY_REDACTION` to `true`.
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:20]: include the http variable also. That is different.

Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:20]: Add a link to OpenTelemetry doc, where it speaks about it.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:20]: thank you, i completely missed that

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,24 @@
 # Release History
 
+## 1.1.1 (2024-04-25)
+
+### Other Changes
+
+* Update OpenTelemetry dependencies.
+  ([#43197](https://github.com/Azure/azure-sdk-for-net/pull/43197))
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:8]: This PR reference is incorrect.

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:8]: Add reference on breaking change section too.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md (hunk 3 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md]: is there a specific doc you want included?

I've included the link to the hotfix itself, as well as the changelogs for Instrumenation.AspNetCore and Instrumentation.Http

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,27 @@
...
+### Other Changes
+
+* Update OpenTelemetry dependencies.
+  ([#43432](https://github.com/Azure/azure-sdk-for-net/pull/43432))
+  - OpenTelemetry 1.8.1
+  - OpenTelemetry.Extensions.Hosting 1.8.1
+  - OpenTelemetry.Instrumentation.AspNetCore 1.8.1
+  - OpenTelemetry.Instrumentation.Http 1.8.1
+  - This includes a hotfix for [GHSA-vh2m-22xx-q94f](https://github.com/open-telemetry/opentelemetry-dotnet/security/advisories/GHSA-vh2m-22xx-q94f) which redacts the query string values by default. 
+### Breaking Changes
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:14]: nit: Add line spacing above breaking changes.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md (hunk 3 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md]: There was a small update made to the AspNetCore and HttpClient instrumentation libraries - https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.AspNetCore

https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.Http

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md]: PR reference - https://github.com/open-telemetry/opentelemetry-dotnet/pull/5532/files

**Comment by mattmccleary** [comment]
My preference is to overwrite the default redaction and continue with the same behavior UNTIL OpenTelemetry makes a spec change. This will minimize the risk of giving our customers whip lash on a behavior that has been standard in Application Insight for most or all of its existence.
CC: @OsvaldoRosado, @SoubhagyaDash

**Comment by TimothyMothra** [comment]
Changing this to Draft. Need to make some changes. Will aim to get this merged by tomorrow (Fri) morning.

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,24 @@
...
+* `OpenTelemetry.Instrumentation.AspNetCore` and `OpenTelemetry.Instrumentation.Http` include a [hotfix](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5532) for [GHSA-vh2m-22xx-q94f](https://github.com/open-telemetry/opentelemetry-dotnet/security/advisories/GHSA-vh2m-22xx-q94f).
+  By default any values detected in the query string component of requests are replaced with the text `Redacted` when building the `url.query` or the `url.full` tag.
+  For example, `?key1=value1&key2=value2` becomes `?key1=Redacted&key2=Redacted`.
+  `Azure.Monitor.OpenTelemetry.AspNetCore` has chosen to disable this redaction until the OpenTelemetry Community has made a decision on the default behavior.
+  ([#43432](https://github.com/Azure/azure-sdk-for-net/pull/43432))
+  You can re-enable this redaction by setting an environment variable:
+    - For `OpenTelemetry.Instrumentation.AspNetCore` set `OTEL_DOTNET_EXPERIMENTAL_ASPNETCORE_DISABLE_URL_QUERY_REDACTION` to `false` before calling `UseAzureMonitor()`.
+    - For `OpenTelemetry.Instrumentation.Http` set `OTEL_DOTNET_EXPERIMENTAL_HTTPCLIENT_DISABLE_URL_QUERY_REDACTION` to `false` before calling `UseAzureMonitor()`.
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:20]: Why is this needed?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:20]: I'm assuming users will seek this hotfix version to mitigate the CVE warning they'll get from their build systems. So I'm trying to explain what the issue is, what was done about it, and how a user can make a change. 

Arguably, since there will be no change in behavior, we could omit all of this.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:20]: We had an internal discussion.
We strongly felt it's important to keep this section explaining Azure Monitor Distro's standpoint.
But, this standpoint could change in the coming weeks.
We will remove this now and explain if there are further changes.

**Comment by mattmccleary** [comment]
Thank you!
